source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -224,6 +228,16 @@ void AuthenticatorImpl::verifyKey() {
     // Remove JWT from headers.
     curr_token_->removeJwt(*headers_);
   }
+  if (set_payload_cb_ && !provider.payload_in_metadata().empty()) {
+    Protobuf::util::JsonParseOptions options;
+    ProtobufWkt::Struct payload_pb;
+    const auto status =
+        Protobuf::util::JsonStringToMessage(jwt_->payload_str_, &payload_pb, options);
+    // payload_str_ have been verified as valid JSON already.
+    // All valid JSON should be able to parse into a portobuf Struct.
```

Comment:
Nit: typo "portobuf" (although I like it).

Reply:
Done.  Thanks